src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -1191,13 +1166,38 @@ public TypeInfo GetTypeByName(string name)
             return res;
         }
 
-        internal void UpdatePdbInformation(Stream streamToReadFrom)
+        internal void UpdatePdbInformationFromSymbolServer(Stream streamToReadFrom)
         {
             var pdbStream = new MemoryStream();
+            streamToReadFrom.Position = 0;
             streamToReadFrom.CopyTo(pdbStream);
+            pdbStream.Position = 0;
             pdbMetadataReader = MetadataReaderProvider.FromPortablePdbStream(pdbStream).GetMetadataReader();
+            if (pdbMetadataReader == null)
+                return;
+            ProcessSourceLink();
+            foreach (var method in this.Methods)
+            {
+                method.Value.pdbMetadataReader = pdbMetadataReader;
+                method.Value.UpdatePdbInformation(method.Value.methodDefHandle);
+            }
         }
-    }
+
+        internal async Task LoadPDBFromSymbolServer(DebugStore debugStore, CancellationToken token)
```

Comment:
We should handle errors with this, so one of the symbol servers failing would not crash the whole thing, or stop other servers from being used.